### PR TITLE
Allow the user to enter package tracking details. Mark III: Yosemite 

### PR DESCRIPTION
**Pull request by ctarda**
This is the third of a few PRs which will add support for allowing users to enter package tracking details manually, with the help of the Shipment Tracking extension in WC.

This PR contains changes to the Yosemite layer.

Ref: #689

Testing

Checkout branch and build. Check that there are no errors in the console (i.e. coredata migration errors)

Run the tests

 If there are user facing changes, I have added an item to RELEASE-NOTES.txt.

### CODE REVIEW

File: Yosemite/YosemiteTests/Stores/ShipmentStoreTests.swift [hunk 1]
```diff
@@ -202,6 +216,148 @@ class ShipmentStoreTests: XCTestCase {
...
+    /// Verifies that `upsertShipmentTrackingProviderData` does not produce duplicate entries.
+    ///
+    func testUpdateRetrieveShipmentTrackingProviderGroupListEffectivelyUpdatesPreexistantShipmentTrackingData() {
+        let shipmentStore = ShipmentStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+        XCTAssertEqual(self.viewStorage.countObjects(ofType: Storage.ShipmentTrackingProviderGroup.self), 0)
+
+        let group = DispatchGroup()
+
+        group.enter()
+        shipmentStore.upsertTrackingProviderDataInBackground(siteID: sampleSiteID,
```
Comment (bummytime) [inline Yosemite/YosemiteTests/Stores/ShipmentStoreTests.swift]: This write to Core Data should probably happen outside of the `DispatchGroup()` — I was seeing a race condition occur while debugging.

See my test branch → https://github.com/woocommerce/woocommerce-ios/blob/try/689-mark3-bummy/Yosemite/YosemiteTests/Stores/ShipmentStoreTests.swift#L285

File: Networking/NetworkingTests/Remote/ShipmentsRemoteTests.swift [hunk 2]
```diff
@@ -370,7 +370,7 @@ final class ShipmentsRemoteTests: XCTestCase {
         let remote = ShipmentsRemote(network: network)
         let expectation = self.expectation(description: "Load shipment tracking providers information")
 
-        network.simulateResponse(requestUrlSuffix: "orders/\(sampleOrderID)/shipment-trackings/providers", filename: "shipment_tracking_providers")
+        network.simulateResponse(requestUrlSuffix: "orders/\(sampleOrderID)/shipment-trackings/providers", filename: "shipmentt")
```
[the remark this answers is not in the document]
Comment (ctarda) [reply Networking/NetworkingTests/Remote/ShipmentsRemoteTests.swift]: 🤦‍♂️

**Comment by ctarda** [comment]
Thanks for unblocking the PR @bummytime !

I incorporated your feedback and added methods to create trackings. I think this is finally ready for a proper review now 😅

**Comment by ctarda** [comment]
Thank *you* @bummytime . I've learnt a lot about CoreData (and about myself 😛 ) working on this PR. 🙇

**Pull request merged**